src/main/java/uk/co/ramp/covid/simulation/Model.java

```diff
@@ -0,0 +1,203 @@
+        return this;
+    }
+
+    public Model setIters(int iters) {
+        this.nIters = iters;
+        return this;
+    }
+
+    public Model setRNGSeed(int seed) {
+        this.rngSeed = seed;
+        return this;
+    }
+
+    public Model setSchoolLockdown(int start, int end, double socialDistance) {
+        this.schoolLockDown = new Lockdown(start, end, socialDistance);
+        return this;
+    }
+
+    public Model setLockdown(int start, int end, double socialDistance) {
+        this.lockDown = new Lockdown(start, end, socialDistance);
+        return this;
+    }
+
+    public Model setOutputFile(String fname) {
+        this.outputFile = fname;
+        return this;
+    }
+
+    public boolean isValid() {
+        boolean valid = true;
+
+        if (populationSize == null) {
+            LOGGER.warn("Uninitialised model parameter: populationSize");
+            valid = false;
+        }
+        if (nIters == null) {
+            LOGGER.warn("Uninitialised model parameter: nIters");
+            valid = false;
+        }
+        if (nHouseholds == null) {
+            LOGGER.warn("Uninitialised model parameter: nHouseholds");
+            valid = false;
+        }
+        if (nInfections == null) {
+            LOGGER.warn("Uninitialised model parameter: nInfections");
+            valid = false;
+        }
+        if (nDays == null) {
+            LOGGER.warn("Uninitialised model parameter: nDays");
+            valid = false;
+        }
+        if (outputFile == null) {
+            LOGGER.warn("Uninitialised model parameter: outputFile");
+            valid = false;
+        }
+        // Handle optional args
+        if (lockDown != null) {
+            LOGGER.warn("lockDown parameters invalid");
+            valid = valid && lockDown.isValid();
+        }
+        if (schoolLockDown != null) {
+            LOGGER.warn("schoolLockDown parameters invalid");
+            valid = valid && schoolLockDown.isValid();
+        }
+
+        return valid;
+    }
+
+    public List<List<DailyStats>> run() {
+        assert isValid() : "Model parameters are invalid";
+
+        List<List<DailyStats>> stats = new ArrayList<>(nIters);
+        for (int i = 0; i < nIters; i++) {
+            Population p = new Population(populationSize, nHouseholds);
+            p.populateHouseholds();
+            p.summarisePop();
+            p.createMixing();
+            p.allocatePeople();
+            p.seedVirus(nInfections);
+            if (lockDown != null) {
+                p.setLockdown(lockDown.start, lockDown.end, lockDown.socialDistance);
+            }
+            if (schoolLockDown != null) {
+                p.setSchoolLockdown(schoolLockDown.start, schoolLockDown.end, schoolLockDown.socialDistance);
+            }
+            stats.add(p.timeStep(nDays));
+        }
+
+        if (outputFile != null) {
+            for (int i = 0; i < nIters; i++) {
+                outputCSV(outputFile, i, stats.get(i));
```

Comment:
There's only one filename, but the outputCSV method is written as though it is writing a new file each time (e.g. creating a new FileWriter, and printing the headers).  What is the intention here?

Reply:
Again looks like temporary code that's snuck in! I'll fix this up now